src/ProductView/ProductView.js

```diff
@@ -95,15 +118,20 @@ class ProductView extends Component {
             this.setState({filter: updatedFilters});
         }
     }
-
-    loadEventsFor(eventType) {
-        this.activeEventType = eventType;
-        this.props.dataSource.fetchEventsOf(
-            this.state.product.id, 
-            eventType.id, 
-            this.handleEventData,
-            this.handleError
-        );
+    
+    setActiveEventType(eventType) {
+        this.setState({activeEventType: eventType});
+    }
+    
+    getCurrentEvents() {
+        const eventContainer = this.state.eventData.find((eventContainer) => {
```

Comment:
The exact same comparison was used in LineChart > handleEventData in line 37. Please extract it to a method.